File: app/src/main/java/com/example/recyclerview/RealMHelper.java
package com.example.recyclerview;

import android.widget.ListView;

import com.example.recyclerview.Model.Tasks;

import java.util.List;

import io.realm.Realm;
import io.realm.RealmConfiguration;
import io.realm.Sort;

public class RealMHelper {

    Realm realm = Realm.getDefaultInstance();



    public void insertData(final Tasks t){
        Number num = realm.where(Tasks.class).max("task_id");
        final int nextID;
        if(num == null){
            nextID = 0;
        }else{
            nextID = num.intValue() + 1;
        }

        t.setTask_id(nextID);
        realm.beginTransaction();
        realm.copyToRealm(t);
        realm.commitTransaction();
    }

    public List<Tasks> getData(){
        return realm.where(Tasks.class).sort("task_id", Sort.DESCENDING).findAll();
    }

    public void editData(final int id, final String task_name){
        final Tasks item = realm.where(Tasks.class).equalTo("task_id", id).findFirst();
        realm.executeTransaction(new Realm.Transaction() {
            @Override
            public void execute(Realm realm) {
                item.setTask_name(task_name);
                realm.copyFromRealm(item);
            }
        });
    }

    public void deleteData(final int id){

        final Tasks item = realm.where(Tasks.class).equalTo("task_id", id).findFirst();
         realm.executeTransaction(new Realm.Transaction() {

             @Override
            public void execute(Realm realm) {

                item.deleteFromRealm();

            }

       });

    }
}

File: app/src/main/java/com/example/recyclerview/UI/RealMHelper.kt
package com.example.recyclerview.UI

import com.example.recyclerview.Model.Tasks
import io.realm.Realm
import io.realm.Sort

class RealMHelper {
    var realm = Realm.getDefaultInstance()
    fun insertData(t: Tasks) {
        val num = realm.where(Tasks::class.java).max("task_id")
        val nextID: Int
        nextID = if (num == null) {
            0
        } else {
            num.toInt() + 1
        }
        t.task_id = nextID
        realm.beginTransaction()
        realm.copyToRealm(t)
        realm.commitTransaction()
    }

    val data: List<Tasks>
        get() = realm.where(Tasks::class.java).sort("task_id", Sort.DESCENDING).findAll()

    fun editData(id: Int, task_name: String?) {
        val item = realm.where(
            Tasks::class.java
        ).equalTo("task_id", id).findFirst()
        realm.executeTransaction { realm ->
            item!!.task_name = task_name
            realm.copyFromRealm(item)
        }
    }

    fun deleteData(id: Int) {
        val item = realm.where(
            Tasks::class.java
        ).equalTo("task_id", id).findFirst()
        realm.executeTransaction { item!!.deleteFromRealm() }
    }
}